### ISSUE TITLE
Element "participating-org" requires @ref or sub-element "narrative" to be populated

### ISSUE THREAD

**Issue by emmajclegg** [younginnovations/iatipublisher#1596]
The IATI Publisher data entry form for activity element `participating-org` currently allows users to save data with both `@/ref` and `narrative` unpopulated:

![image](https://github.com/user-attachments/assets/989d2d82-583c-44b6-b3b3-d5573dafe3c6)

This would be very easy to do and will cause errors in published data as, according to the Standard, _"the participating organisation must have an identifier or a narrative"_

1. Can we add data validation to the data entry form to force the user to populate one of these fields before saving.
2. Can we rename "narrative" to "name" in the interface (similar to what we did for reporting-org in #1406) to make it clearer to the user what this field is.
3. Can we (ideally) move the `narrative` sub-element to the top of the form as it is logical to the user to complete this first.

### TODO
- [x] Can we add data validation to the data entry form to force the user to populate one of these fields before saving.
- [x] Can we rename "narrative" to "name" in the interface (similar to what we did for reporting-org in #1406) to make it clearer to the user what this field is.
- [x] Can we (ideally) move the `narrative` sub-element to the top of the form as it is logical to the user to complete this first.

**Comment by PG-Momik** [comment]
@emmajclegg Question on this. 
For imports, should we not let users import activity if both  participating org reference and narrative (name) is not provided?

**Comment by emmajclegg** [comment]
> For imports, should we not let users import activity if both participating org reference and narrative (name) is not provided?

@PG-Momik - yes, this should apply to imports too ideally. Provided this is not too much effort to implement for xls though, considering we will be changing that import process soon

**Comment by Sanam-07** [comment]
- [ ]  Issue 1 : The error message occurs when the user edit the participating Organization and inverse the filled name and reference.
![image](https://github.com/user-attachments/assets/1ac17239-aed6-4a9b-aac2-127fa79baf4a)
<-----------------------------Video------------------------------->
[Untitled_ Nov 13,2024 4_56 PM.webm](https://github.com/user-attachments/assets/e77a63dc-a965-44a1-8062-4f08c069f871)

**Comment by emmajclegg** [comment]
Thanks @Sanam-07 

The data entry form check looks good and the change you've made to the logic for calculating completeness makes sense.

> Additionally, a required (*) indicator will appear next to the reference and name fields.

Some organisations will not have a `reference` for users to enter, so please can we remove the required (*) against this field. All organisations should have a `narrative` (i.e. name), so it is reasonable for this to be marked as a required field to help the user complete the data entry form.

> During imports, an error message of the 'error' level will be displayed if the conditions are not met.

To check normal behaviour, are error messages displayed for all import file types (xml, csv and xlsx), or just xlsx?

I see an error message for missing `participating-org name` and `reference` when I've tested an xlsx file import, but not for an equivalent csv import (e.g. [Test csv import - organisation _Emma's test publisher - Sep 2024.csv](https://github.com/user-attachments/files/17737602/Test.csv.import.-.organisation._Emma.s.test.publisher.-.Sep.2024.csv))

**Comment by PG-Momik** [comment]
@emmajclegg 

> To check normal behavior, are error messages displayed for all import file types (xml, csv and xlsx), or just xlsx?

No, the error message should be displayed for all import types. I've redeployed the changes + the new changes with required(*) removed from reference. Please let me know if the issue persists.

**Comment by emmajclegg** [comment]
Thanks @PG-Momik - the csv import issue has been resolved.

A last minor comment - I've noticed that only the `reference` field is highlighted with a red border when the error messages display. I think name should also be highlighted red, given the error message relates to both fields.

![Image](https://github.com/user-attachments/assets/c9f3e9f5-26b8-49da-b66f-0a75dedeed79)

Once that's updated, I'm happy for this issue's changes to be deployed.
